telegram/src/main/kotlin/com/github/kotlintelegrambot/network/multipart/MultipartBodyFactory.kt

```diff
@@ -44,22 +41,21 @@ internal class MultipartBodyFactory(private val gson: Gson) {
         return listOfNotNull(chatIdPart, mediaGroupPart, disableNotificationPart, replyToMessageIdPart, allowSendingWithoutReplyPart) + filesParts
     }
 
-    private fun MediaGroup.takeFiles(): List<Pair<File, String>> = medias.flatMap { groupableMedia ->
-        when {
-            groupableMedia is InputMediaPhoto && groupableMedia.media is TelegramFile.ByFile -> listOf(
-                groupableMedia.media.file to MediaTypeConstants.IMAGE
-            )
-            groupableMedia is InputMediaVideo && groupableMedia.media is TelegramFile.ByFile && groupableMedia.thumb != null -> listOf(
-                groupableMedia.media.file to MediaTypeConstants.VIDEO,
-                groupableMedia.thumb.file to MediaTypeConstants.IMAGE
-            )
-            groupableMedia is InputMediaVideo && groupableMedia.media is TelegramFile.ByFile -> listOf(
-                groupableMedia.media.file to MediaTypeConstants.VIDEO
-            )
-            groupableMedia is InputMediaVideo && groupableMedia.thumb != null -> listOf(
-                groupableMedia.thumb.file to MediaTypeConstants.IMAGE
-            )
-            else -> emptyList()
+    private fun MediaGroup.takeFilesParts(): List<MultipartBody.Part> = medias.flatMap { groupableMedia ->
+        val mediaType = when (groupableMedia) {
+            is InputMediaPhoto -> MediaTypeConstants.IMAGE
+            is InputMediaVideo -> MediaTypeConstants.VIDEO
+            is InputMediaAudio -> MediaTypeConstants.AUDIO
+            else -> null
```

Comment:
Do you think it would be better to be exhaustive here and add all the possible branches here? That way, if a new `GroupableMedia` is added, this code won't compile and whoever is doing the changes will be forced to consider how the new type should be dealt with here.

Reply:
Good point. Fixed.